### PR TITLE
drivers: stepper: add invert direction to gpio stepper

### PR DESCRIPTION
**Pull request by jilaypandya**
This commit implements invert direction feature in gpio stepper

### CODE REVIEW

**Comment by bjarki-andreasen** [comment]
looks good, but could you split the renaming of the compatible, and the inclusion of the new property into separate commits?

**Comment by decsny** [comment]
you don't need to fight the bot, it puts DT label for intended reason

File: drivers/stepper/gpio_stepper_controller.c [hunk 1]
```diff
@@ -326,6 +345,7 @@ static int gpio_stepper_motor_controller_init(const struct device *dev)
 		ARRAY_SIZE(gpio_stepper_motor_control_pins_##child) == 4,                          \
 		"gpio_stepper_controller driver currently supports only 4 wire configuration");    \
 	static const struct gpio_stepper_config gpio_stepper_config_##child = {                    \
+		.invert_direction = DT_PROP(child, invert_direction),                              \
```
Comment (dipakgmx) [inline drivers/stepper/gpio_stepper_controller.c:348]:
```suggestion
		.invert_direction = DT_PROP_OR(child, invert_direction, 0),                              \
```

Comment (jilaypandya) [reply drivers/stepper/gpio_stepper_controller.c:348]: Boolean properties are true if present. They should not have a value. A boolean property is only false if it is completely missing in the DTS.

**Review by dipakgmx — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]
nice :)

File: drivers/stepper/gpio_stepper_controller.c [hunk 2]
```diff
@@ -53,20 +55,35 @@ static int stepper_motor_set_coil_charge(const struct device *dev)
...
 }
 
+static void increment_coil_charge(const struct device *dev)
+{
+	struct gpio_stepper_data *data = dev->data;
+
+	data->coil_charge =
+		data->coil_charge == NUM_CONTROL_PINS * MAX_MICRO_STEP_RES - data->step_gap
+			? 0
+			: data->coil_charge + data->step_gap;
```
Comment (fabiobaltieri) [inline drivers/stepper/gpio_stepper_controller.c:65]: yikes, can you expand it to a normal `if`? the indentation is something else here

File: drivers/stepper/gpio_stepper_controller.c [hunk 3]
```diff
@@ -53,20 +55,35 @@ static int stepper_motor_set_coil_charge(const struct device *dev)
...
+}
+
+static void decrement_coil_charge(const struct device *dev)
+{
+	struct gpio_stepper_data *data = dev->data;
+
+	data->coil_charge = data->coil_charge == 0
+				    ? NUM_CONTROL_PINS * MAX_MICRO_STEP_RES - data->step_gap
+				    : data->coil_charge - data->step_gap;
```
Comment (fabiobaltieri) [inline drivers/stepper/gpio_stepper_controller.c:74]: same

**Review by fabiobaltieri — CHANGES_REQUESTED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by dipakgmx — APPROVED** [review]

**Pull request merged**